client/test/router_store.spec.js

```diff
@@ -188,401 +173,500 @@ describe('Router Store', () => {
-      assert.isFalse(routerStore.isUnread(map(fixtures.message1)));
-      assert.isTrue(routerStore.isUnread(map(fixtures.message3)));
-    });
-    it('isFlagged', () => {
-      assert.isFalse(routerStore.isFlagged(map(fixtures.message1)));
-      assert.isTrue(routerStore.isFlagged(map(fixtures.message3)));
-    });
-    it('isAttached', () => {
-      assert.isFalse(
-        routerStore.isAttached(map(fixtures.message1)));
-      assert.isTrue(
-        routerStore.isAttached(map(fixtures.message3)));
-    });
-    it('isDeleted', () => {
-      changeRoute({ });
-      assert.isFalse(
-        routerStore.isDeleted(map(fixtures.message1)));
-      assert.isTrue(
-        routerStore.isDeleted(map(fixtures.message3)));
-    });
-    it('isDraft', () => {
-      assert.isFalse(
-        routerStore.isDraft(map(fixtures.message1)));
-      assert.isTrue(
-        routerStore.isDraft(map(fixtures.message4)));
-    });
-    it('getMailboxTotal', () => {
-      assert.equal(
-        routerStore.getMailboxTotal(),
-        fixtures.account.mailboxes[0].nbTotal
-      );
-      changeRoute({ });
-      assert.equal(
-        routerStore.getMailboxTotal(),
-        fixtures.account.mailboxes[0].nbTotal
-      );
-      changeRoute({ flags: MessageFilter.UNSEEN });
-      assert.equal(
-        routerStore.getMailboxTotal(),
-        fixtures.account.mailboxes[0].nbUnread
-      );
-      changeRoute({ flags: MessageFilter.FLAGGED });
-      assert.equal(
-        routerStore.getMailboxTotal(),
-        fixtures.account.mailboxes[0].nbFlagged
-      );
-    });
-    it('hasNextPage', () => {
-      dispatcher.dispatch({
-        type: ActionTypes.MESSAGE_FETCH_SUCCESS,
-        value: {
-          lastPage: {
-            info: 'last-page',
-            isComplete: false,
-          },
-          result: {},
-          timestamp: new Date(),
-        },
+    // TODO: tester lorsqu'on ne trouve aucun comptes
+    describe('AccountStore', () => {
+
+      it('Select Mailbox', () => {
+        const accountID = fixtures.account.id;
+        const mailboxID = fixtures.account.inboxMailbox;
+        dispatcher.dispatch({
+          type: ActionTypes.ROUTE_CHANGE,
+          value: {
+            accountID: fixtures.account.id,
+            mailboxID: fixtures.account.inboxMailbox,
+          }
+        });
+
+        assert.equal(routerStore.getAccountID(), accountID);
+        assert.equal(routerStore.getAccount().get('id'), accountID);
+
+        assert.equal(routerStore.getMailboxID(), mailboxID);
+        assert.equal(routerStore.getMailbox().get('id'), mailboxID);
       });
-      assert.isTrue(routerStore.hasNextPage());
-      dispatcher.dispatch({
-        type: ActionTypes.MESSAGE_FETCH_SUCCESS,
-        value: {
-          lastPage: {
-            info: 'last-page',
-            isComplete: true,
-          },
-          result: {},
-          timestamp: new Date(),
-        },
+
+
+      it('Select Tab from AccountEdit page', () => {
```

Comment:
We have a question here about whether these tests should be considered as unit tests or functional tests. The ROUTE_CHANGE event looks tricky to test and requires more functional testing than other events. 
But, it raises another question, should testing events be considered as unit tests of functional tests?

Whatever the answer to that questions do not have any consequence on what you are doing. I mean, we should discuss later of the file organization of the email tests.